Foreground use cases in the Identity Security docs

ptgott opened the pull request:
This change cleans up the Identity Security docs by providing a more self-evident hierarchy of pages that makes it possible to place use case information prominently:

- **Clean up the sidebar by organizing it into a set of clear themes:** (a) using Identity Security in the Web UI; (b) enabling integrations, (c) using Session Recording Summaries (which are outside the Identity Security UI), and for more niche use cases, (d) self-hosting Teleport Identity Security (which appears last in the sidebar).

- **Add a usage section that mirrors the organization of the Web UI.** This allows us to have a dedicated page for each Web UI view in the Identity Security tab with use case information displayed prominently.

  Currently, the Identity Security docs bear little resemblance to the structure of the Web UI, so Web UI users need to do more work to find the right information. Note that some links in the Identity Security Web UI tab are currently missing from the docs. The goal of this change is not to introduce new documentation, but to organize existing docs and add prominent use case information.

  As part of this work, combine all Graph Explorer content into a single page (it used to be distributed across the Using and Connections pages).

- **Separate deployment and usage-specific terms:** The terms "Access Graph" and "Identity Activity Center" are nowhere to be found in the Web UI, and only relevant for self-hosted users. For users who want to use Identity Security in the Web UI, mentioning these terms outside the self-hosting section is potentially confusing. Make sure these terms are only present in the self-hosting section.

  Since "Access Graph" is legacy terminology for a Web UI view (now Graph Explorer), it is still used throughout the Integrations section. This change only focuses on section index pages and introductory paragraphs to limit the scope to the most visible places.

ptgott committed:
Foreground use cases in the Identity Security docs

This change cleans up the Identity Security docs by providing a more
self-evident hierarchy of pages that makes it possible to place use case
information prominently:

- **Clean up the sidebar by organizing it into a set of clear themes:**
  (a) using Identity Security in the Web UI; (b) enabling integrations,
  (c) using Session Recording Summaries (which are outside the Identity
  Security UI), and for more niche use cases, (d) self-hosting Teleport
  Identity Security (which appears last in the sidebar).

- **Add a usage section that mirrors the organization of the Web UI.**
  This allows us to have a dedicated page for each Web UI view in the
  Identity Security tab with use case information displayed prominently.

  Currently, the Identity Security docs bear little resemblance to the
  structure of the Web UI, so Web UI users need to do more work to find
  the right information. Note that some links in the Identity Security
  Web UI tab are currently missing from the docs. The goal of this
  change is not to introduce new documentation, but to organize existing
  docs and add prominent use case information.

  As part of this work, combine all Graph Explorer content into a single
  page (it used to be distributed across the Using and Connections
  pages).

- **Separate deployment and usage-specific terms:** The terms "Access
  Graph" and "Identity Activity Center" are nowhere to be found in the
  Web UI, and only relevant for self-hosted users. For users who want to
  use Identity Security in the Web UI, mentioning these terms outside
  the self-hosting section is potentially confusing. Make sure these
  terms are only present in the self-hosting section.

  Since "Access Graph" is legacy terminology for a Web UI view (now
  Graph Explorer), it is still used throughout the Integrations section.
  This change only focuses on section index pages and introductory
  paragraphs to limit the scope to the most visible places.

r0mant commented on docs/pages/identity-security/access-graph/access-graph.mdx:
```diff
@@ -1,14 +1,23 @@
-Teleport Identity Security) requires running the Access Graph Service on your own
-infrastructure. The following guides show you how to deploy the Access Graph
-Service.
+If you run a self-hosted Teleport cluster, using Teleport Identity Security
+requires running two services on your own infrastructure:
+
+- **Teleport Access Graph Service:** Powers access path exploration features,
+  including the Dashboard, Browse, Crown Jewels, Graph Explorer, and SQL Editor
+  views.
+- **Teleport Identity Activity Center:** Powers audit log centralization and
+  querying, including the Browse and Alerts views.
```
@ptgott These are not two separate services you need to run. Identity Activity Center is enabled via configuration in the same service that provides Access Graph. This reads a bit confusing making it seem like you have to run 2 things.

ptgott committed:
Improve Identity Security documentation

Respond to zmb3 feedback:

- Remove inaccurate statement about the Identity Security sidebar item.
- Mention AI in the Session Recording Summaries sidebar entry.
- Add a more representative meta-description to the SSH Key Scanning
  page.

Respond to r0mant feedback: clarify that Identity Activity Center is not
a separate service, but a configuration option of the Access Graph
Service that requires supporting infrastructure.
